src/org.openjdk.nashorn/share/classes/org/openjdk/nashorn/internal/lookup/MethodHandleFactory.java

```diff
@@ -369,13 +369,11 @@ protected static String describe(final Object... data) {
         }
 
         public MethodHandle debug(final MethodHandle master, final String str, final Object... args) {
-            if (log.isEnabled()) {
-                if (PRINT_STACKTRACE) {
-                    stacktrace(log);
-                }
-                return addDebugPrintout(log, Level.INFO, master, Integer.MAX_VALUE, false, str + ' ' + describe(args));
+            // The static `log` field can be `null` if the `MethodHandlerFactory` class has not been initialized yet.
```

Comment:
Don't you need the null check here then too?

Reply:
Unless I missed some code path, the `addDebugPrintout` and `stacktrace` both handle a `null` input correctly.

https://github.com/openjdk/nashorn/blob/7bd13f5684711f60d838229f594874535336f370/src/org.openjdk.nashorn/share/classes/org/openjdk/nashorn/internal/lookup/MethodHandleFactory.java#L282-L287

https://github.com/openjdk/nashorn/blob/7bd13f5684711f60d838229f594874535336f370/src/org.openjdk.nashorn/share/classes/org/openjdk/nashorn/internal/lookup/MethodHandleFactory.java#L194-L204